### ISSUE TITLE
[python] DataFrame value_filter failing on bool column

### ISSUE THREAD

**Issue by bkmartinjr** [single-cell-data/TileDB-SOMA#576]
A value filter (query condition) on a bool-typed column fails.  It works fine on a string column.

Example:
```python
In [34]: census = soma.Collection("s3://cellxgene-data-public/cell-census/2022-12-01/soma", ctx=tiledb.Ctx(config={"vfs.s3.region": "us-west-2"}))

In [35]: census['census_data']['mus_musculus'].obs.read_as_pandas_all(column_names=['soma_joinid', 'is_primary_data', 'cell_type'])
Out[35]: 
         soma_joinid  is_primary_data                cell_type
0                  0             True          oligodendrocyte
1                  1             True          oligodendrocyte
2                  2             True          oligodendrocyte
3                  3             True          oligodendrocyte
4                  4             True          oligodendrocyte
...              ...              ...                      ...
3409714      3409714             True  cerebellar granule cell
3409715      3409715             True  cerebellar granule cell
3409716      3409716             True  cerebellar granule cell
3409717      3409717             True  cerebellar granule cell
3409718      3409718             True  cerebellar granule cell

[3409719 rows x 3 columns]

In [36]: census['census_data']['mus_musculus'].obs.read_as_pandas_all(column_names=['soma_joinid', 'is_primary_data', 'cell_type'], value_filter="is_primary_data == True")
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[36], line 1
----> 1 census['census_data']['mus_musculus'].obs.read_as_pandas_all(column_names=['soma_joinid', 'is_primary_data', 'cell_type'], value_filter="is_primary_data == True")

File ~/TileDB-SOMA/apis/python/src/tiledbsoma/dataframe.py:388, in DataFrame.read_as_pandas_all(self, ids, value_filter, column_names, result_order)
    381 def read_as_pandas_all(
    382     self,
    383     ids: Optional[SparseDataFrameCoordinates] = None,
   (...)
    386     result_order: Optional[ResultOrder] = None,
    387 ) -> pd.DataFrame:
--> 388     return pd.concat(
    389         self.read_as_pandas(
    390             ids=ids,
    391             value_filter=value_filter,
    392             column_names=column_names,
    393             result_order=result_order,
    394         ),
    395         ignore_index=True,
    396     )

File ~/cell-census/venv/lib/python3.9/site-packages/pandas/util/_decorators.py:331, in deprecate_nonkeyword_arguments.<locals>.decorate.<locals>.wrapper(*args, **kwargs)
    325 if len(args) > num_allow_args:
    326     warnings.warn(
    327         msg.format(arguments=_format_argument_list(allow_args)),
    328         FutureWarning,
    329         stacklevel=find_stack_level(),
    330     )
--> 331 return func(*args, **kwargs)

File ~/cell-census/venv/lib/python3.9/site-packages/pandas/core/reshape/concat.py:368, in concat(objs, axis, join, ignore_index, keys, levels, names, verify_integrity, sort, copy)
    146 @deprecate_nonkeyword_arguments(version=None, allowed_args=["objs"])
    147 def concat(
    148     objs: Iterable[NDFrame] | Mapping[HashableT, NDFrame],
   (...)
    157     copy: bool = True,
    158 ) -> DataFrame | Series:
    159     """
    160     Concatenate pandas objects along a particular axis.
    161 
   (...)
    366     1   3   4
    367     """
--> 368     op = _Concatenator(
    369         objs,
    370         axis=axis,
    371         ignore_index=ignore_index,
    372         join=join,
    373         keys=keys,
    374         levels=levels,
    375         names=names,
    376         verify_integrity=verify_integrity,
    377         copy=copy,
    378         sort=sort,
    379     )
    381     return op.get_result()

File ~/cell-census/venv/lib/python3.9/site-packages/pandas/core/reshape/concat.py:422, in _Concatenator.__init__(self, objs, axis, join, keys, levels, names, ignore_index, verify_integrity, copy, sort)
    420     objs = [objs[k] for k in keys]
    421 else:
--> 422     objs = list(objs)
    424 if len(objs) == 0:
    425     raise ValueError("No objects to concatenate")

File ~/TileDB-SOMA/apis/python/src/tiledbsoma/dataframe.py:373, in DataFrame.read_as_pandas(self, ids, value_filter, column_names, result_order)
    366 def read_as_pandas(
    367     self,
    368     ids: Optional[SparseDataFrameCoordinates] = None,
   (...)
    371     result_order: Optional[ResultOrder] = None,
    372 ) -> Iterator[pd.DataFrame]:
--> 373     for tbl in self.read(
    374         ids=ids,
    375         value_filter=value_filter,
    376         column_names=column_names,
    377         result_order=result_order,
    378     ):
    379         yield tbl.to_pandas()

File ~/TileDB-SOMA/apis/python/src/tiledbsoma/dataframe.py:222, in DataFrame.read(self, ids, value_filter, column_names, result_order)
    217 if result_order is not None and result_order not in get_args(ResultOrder):
    218     raise ValueError(
    219         "result_order must be one of " + ", ".join(get_args(ResultOrder))
    220     )
--> 222 sr = clib.SOMAReader(
    223     self._uri,
    224     name=self.__class__.__name__,
    225     schema=A.schema,  # query_condition needs this
    226     column_names=column_names,
    227     query_condition=query_condition,
    228     platform_config={} if self._ctx is None else self._ctx.config().dict(),
    229     result_order=(result_order or "auto"),
    230 )
    232 if ids is not None:
    233     if not isinstance(ids, (list, tuple)):

RuntimeError: TileDBError: PyQueryCondition.init_bool() not found.

At:
  /home/bruce/TileDB-SOMA/apis/python/src/tiledbsoma/query_condition.py(384): init_pyqc
  /home/bruce/TileDB-SOMA/apis/python/src/tiledbsoma/query_condition.py(244): aux_visit_Compare
  /home/bruce/TileDB-SOMA/apis/python/src/tiledbsoma/query_condition.py(196): visit_Compare
  /usr/lib/python3.9/ast.py(407): visit
  /home/bruce/TileDB-SOMA/apis/python/src/tiledbsoma/query_condition.py(133): init_query_condition
  /home/bruce/TileDB-SOMA/apis/python/src/tiledbsoma/dataframe.py(222): read
  /home/bruce/TileDB-SOMA/apis/python/src/tiledbsoma/dataframe.py(373): read_as_pandas
  /home/bruce/cell-census/venv/lib/python3.9/site-packages/pandas/core/reshape/concat.py(422): __init__
  /home/bruce/cell-census/venv/lib/python3.9/site-packages/pandas/core/reshape/concat.py(368): concat
  /home/bruce/cell-census/venv/lib/python3.9/site-packages/pandas/util/_decorators.py(331): wrapper
  /home/bruce/TileDB-SOMA/apis/python/src/tiledbsoma/dataframe.py(388): read_as_pandas_all
  <ipython-input-36-dd781461e300>(1): <module>
  /home/bruce/cell-census/venv/lib/python3.9/site-packages/IPython/core/interactiveshell.py(3433): run_code
  /home/bruce/cell-census/venv/lib/python3.9/site-packages/IPython/core/interactiveshell.py(3373): run_ast_nodes
  /home/bruce/cell-census/venv/lib/python3.9/site-packages/IPython/core/interactiveshell.py(3194): run_cell_async
  /home/bruce/cell-census/venv/lib/python3.9/site-packages/IPython/core/async_helpers.py(129): _pseudo_sync_runner
  /home/bruce/cell-census/venv/lib/python3.9/site-packages/IPython/core/interactiveshell.py(2995): _run_cell
  /home/bruce/cell-census/venv/lib/python3.9/site-packages/IPython/core/interactiveshell.py(2940): run_cell
  /home/bruce/cell-census/venv/lib/python3.9/site-packages/IPython/terminal/interactiveshell.py(678): interact
  /home/bruce/cell-census/venv/lib/python3.9/site-packages/IPython/terminal/interactiveshell.py(685): mainloop
  /home/bruce/cell-census/venv/lib/python3.9/site-packages/IPython/terminal/ipapp.py(318): start
  /home/bruce/cell-census/venv/lib/python3.9/site-packages/traitlets/config/application.py(985): launch_instance
  /home/bruce/cell-census/venv/lib/python3.9/site-packages/IPython/__init__.py(123): start_ipython
  /home/bruce/cell-census/venv/bin/ipython(8): <module>

```

**Comment by johnkerl** [comment]
PR #579 needs to wait for core 2.13.0 and TileDB-Py 0.19.0 which should be this week.

**Comment by bkmartinjr** [comment]
@johnkerl - this is resolved, correct?

**Comment by johnkerl** [comment]
@bkmartinjr yes, sorry, in a conference call at the moment but will update soon

**Comment by johnkerl** [comment]
For posterity: this fix required #576 as well as core 2.13 support (#587).
